### ISSUE TITLE
Adding a “Tape” tool

### ISSUE THREAD

**Issue by MiliAxe** [xournalpp/xournalpp#5446]
### Is Your Feature Request Related to a Problem?

Many students might need to quickly take a quiz of themselves while reading content from a book. Not seeing the answer to a question and answering it from memory can be a vital tool in learning/memorizing content. 

### Feature Description

It allows you to add tape over content and it enables you to quickly reveal the content underneath it by clicking on it. 

![IMG_2890](https://github.com/xournalpp/xournalpp/assets/55515322/f9aedcde-a374-40d6-9336-b279a84100c7)

This ensures a very easy way to take quizzes of yourself with minimal effort. 

### Alternative Solutions

If this feature is considered too much of a “bloat” to be included in the program, an alternative approach would be to develop a specific extension for this program. I would appreciate it if anyone can tell if this is technically possible to do with the extensions.

**Comment by frukto** [comment]
+1 I always try to simulate this with layers, but it is very cumbersome

**Comment by atticus-sullivan** [comment]
> +1 I always try to simulate this with layers, but it is very cumbersome

How exactly do you simulate this with layers? Maybe we can do something with a plugin.

**Comment by MiliAxe** [comment]
> > +1 I always try to simulate this with layers, but it is very cumbersome
> 
> How exactly do you simulate this with layers? Maybe we can do something with a plugin.

I cover the material I want with a rectangle box in a separate layer:

![image](https://github.com/xournalpp/xournalpp/assets/55515322/45269850-d556-458c-a9a2-9d38f85692b9)

After I'm done, I can disable the top layer to see the content beneath it.

![image](https://github.com/xournalpp/xournalpp/assets/55515322/b454ab96-9e6c-42b2-9067-94749c4b224d)

**Comment by frukto** [comment]
Yes, I actually have "solutions" layer containing the solutions. I don't cover anything with boxes, I rather completely blend out content.
However it is inconvenient to handle many layers to not have to show all stuff at once.
MiliAxe's tapes would be much more handy.

However I don't know how to handle such tapes in pdf export. Maybe just use an extra layer for all of them.

**Comment by atticus-sullivan** [comment]
Why not simply use a single layer for the "tapes" and delete them via the delete stroke function of the eraser? Since it's a different layer, you won't accidentally remove anything of your writing and with the delete stroke you only have to click on the "tape" once.

For ease of use, we could write a plugin which registers a new toolbar icon. This could automatically activate the eraser, switch to delete stroke and go to the "tape layer". Similar thing for a pen icon for creating the tapes.

**Comment by T4ForFun** [comment]
I think with this being a feature in notability, a very big app for note-taking it would not be too specific for the base app. At the moment i am exporting my notes to pdf to use them in another app, so this feature would be much appreciated

**Comment by atticus-sullivan** [comment]
I'm still wondering if we can't use just a layer for this like shown in the following:

![Peek 2024-06-18 23-51](https://github.com/xournalpp/xournalpp/assets/60581958/0349a851-663b-4716-8246-b6949788c935)

Isn't this already pretty close to the workflow with tape-feature of notability?
(the extra layer for the "tape" prevents accidentally deleting something of the actual content)
